Dockerfile-debian.template

```diff
@@ -17,19 +17,34 @@ RUN set -eux; \
 # usage: COPY --from=dart:xxx /runtime/ /
 # uses hard links here to save space
 RUN set -eux; \
-    for f in \
+    case "$(uname -m)" in \
```

Comment:
Seems like docker official images maintainers prefer `dpkg --print-architecture` over `uname`:

https://github.com/docker-library/official-images#multiple-architectures
```
If different parts of the Dockerfile only happen in one architecture or another, use control flow (e.g.if/case) along with dpkg --print-architecture or apk -print-arch to detect the userspace architecture. Only use uname for architecture detection when more accurate tools cannot be installed.
```

Reply:
I went with `uname -m` rather than `dpkg --print-architecture` because it's more precise about the differences between armv6 and armv7 (and this was something I spoke to @justincormack about in my early days of trying to get this working).

> Only use `uname` for architecture detection when more accurate tools cannot be installed.

Seems a little contrary to reality here, as `uname -m` is more accurate.

Raspberry Pi 1 running Raspbian (32 bit):

```
$ dpkg --print-architecture
armhf
$ uname -m
armv6l
```
Raspberry Pi 4 running Raspbian (32 bit): 

```
$ dpkg --print-architecture
armhf
$ uname -m
armv7l
```

Raspberry Pi 4 running Ubuntu (64 bit): 

```
$ dpkg --print-architecture
arm64
$ uname -m
aarch64
```

Reply:
The problem I see with `dpkg --print-architecture` is that it give the same answer of `armhf` for both armv6l and armv7l, where armv7 is supported by Dart and armv6 isn't any longer. So people with earlier Raspberry Pis who might try to run `docker build` against the Dockerfile will get much further into the build process before the wheels come off, rather than an early error stating that their architecture isn't supported.

Reply:
There's no armv6 variant of Debian's buster image on Docker Hub. So wouldn't folks on an earlier Raspberry Pi use arm32v5/debian which we don't support and the build would see `armel` and bail out?

Reply:
For the purpose of this PR, I consider "Dart doesn't support armv6.5" as a fact (if that should change, the right place for the discussion would be the Dart SDK issue tracker). Given that there was https://github.com/dart-lang/sdk/issues/42069 it's unlikely to change (it does mention the Pi Zero).

Regarding `uname` vs. `dpkg`, I will use the current advice from the official images documentation to avoid friction when we update the dart library. I'm happy to switch this to `uname` or anything else whenever the documentation gets updated. This may have to be revisited if we want to use a different base distribution.